devenv/sync.py

```diff
@@ -30,7 +31,9 @@ def main(context: Context, argv: Sequence[str] | None = None) -> int:
         "repo": repo.name,
         "coderoot": context.get("code_root"),
     }
-    return module.main(context_compat)  # type: ignore
+
+    with contextlib.chdir(repo.path):
```

Comment:
@asottile-sentry can i drop 3.10 from tox? not sure why you wanted to keep it around, the only comment says "for internal pypi compat"

Reply:
it looks like ops:salt still uses python 3.10 otherwise we could drop it from pypi I think

Reply:
sync.py is run with devenv's python 3.11 (see install-devenv.sh), that 3.10 venv is only used for salt